### PR TITLE
Added _eval_is_commutative method in stats/symbolic_probability.py

### LINKED ISSUE #24132: print fails with MatMul and Expectation.
The following code worked in 1.10.1 but not in 1.11.1.

```python
from sympy import *
from sympy.stats import Expectation
from sympy.stats.rv import RandomSymbol

print(RandomSymbol("ve") * Identity(2)) # works
z = Expectation(RandomSymbol("ve")) * Identity(2) # works
print(z) # NotImplementedError

```

I have tracked down this error to the cause that
MatMul.could_extract_minus_sign exists in 1.10.1 but not in 1.11.1 by
the commit below. I don't know previous bugs of the commit.

https://github.com/sympy/sympy/commit/1d3c89501c65e4b6cd3b635be60ba1d2bf003b4d

### PR DESCRIPTION
**Pull request by ghost**
Added _eval_is_commutative method to recursively check if its argument is commutative.

#### References to other Issues or PRs

Fixes #24132 

#### Brief description of what is fixed or changed

Example:
In sympy version 1.11.1 for the following code:
```
from sympy import *
from sympy.stats import Expectation
from sympy.stats.rv import RandomSymbol

print(RandomSymbol("ve") * Identity(2)) # worked
z = Expectation(RandomSymbol("ve")) * Identity(2) # worked
print(z) # NotImplementedError was shown here while printing
```
is_commutative should not be None which indirectly refers to false and while using Expectation for the above code it raised NotImplementedError: noncommutative scalars in MatMul are not supported.

Therefore, _eval_is_commutative method now will recursively check if its argument is commutative which is alternative way to set is_commutative = True on Expectation.

#### Other comments

- [x] Opened on a seperate branch
- [x] Follows Documentation Guidelines
- [x] Has Release Notes

#### Release Notes

* stats
  * Fixed failing of print with MatMul and Expectation

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
If `is_commutative` should always be true for a particular class then it's better to set the class variable `is_commutative=True` directly rather than by using `_eval_is_commutative`.

File: sympy/stats/symbolic_probability.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class Probability(Expr):
     """
     def __new__(cls, prob, condition=None, **kwargs):
         prob = _sympify(prob)
+        is_commutative = True
```
Comment (KuldeepBorkar) [inline sympy/stats/symbolic_probability.py:64]: @oscarbenjamin @NikhilSDate `is_commutative = True` was supposed to be added this way or this was needed to be done in some other way and I made a mistake ; ( ??

Comment (KuldeepBorkar) [reply sympy/stats/symbolic_probability.py]: Done

**Comment by oscarbenjamin** [comment]
Tests should be added for what is fixed.

Also it looks like some doctests have failed because printing order changed for some expressions. It should be fine just to change those doctests.

**Comment by oscarbenjamin** [comment]
I've marked this as draft. Feel free to mark it as not draft when it is ready for review.

**Comment by oscarbenjamin** [comment]
Closing n favour of gh-24540

**Pull request closed without merging**